### ISSUE TITLE
IllegalStateException: MQTT endpoint is closed

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#2750]
If an MQTT device aborts its connection attempt before having received a CONNACK, there may be this exception in the MQTT adapter:
````
java.lang.IllegalStateException: MQTT endpoint is closed
	at io.vertx.mqtt.impl.MqttEndpointImpl.checkClosed(MqttEndpointImpl.java:724)
	at io.vertx.mqtt.impl.MqttEndpointImpl.sslSession(MqttEndpointImpl.java:704)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.reportFailedConnectionAttempt(AbstractVertxBasedMqttProtocolAdapter.java:446)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.lambda$handleEndpointConnection$13(AbstractVertxBasedMqttProtocolAdapter.java:432)
	at io.vertx.core.Future.lambda$onFailure$1(Future.java:166)
	at io.vertx.core.impl.FutureImpl.dispatch(FutureImpl.java:105)
	at io.vertx.core.impl.FutureImpl.onComplete(FutureImpl.java:83)
	at io.vertx.core.Future.onFailure(Future.java:164)
	at org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter.handleEndpointConnection(AbstractVertxBasedMqttProtocolAdapter.java:423)
	at io.vertx.mqtt.impl.MqttServerConnection.handleConnect(MqttServerConnection.java:289)
	at io.vertx.mqtt.impl.MqttServerConnection.handleMessage(MqttServerConnection.java:108)
	at io.vertx.mqtt.impl.MqttServerImpl.lambda$null$3(MqttServerImpl.java:102)
````
This leads to the `CONNECT` tracing span to not get finished. Also the metrics reporting for this request won't be done.

**Comment by calohmn** [comment]
Fixed via 60ec983c7c2977d70f0c5b64d1cc53764fcb8eff.

**Comment by sophokles73** [comment]
@calohmn looks like this should also go into the current bug fix branches, right?

**Comment by calohmn** [comment]
I've applied the fix to 1.8.x now (573146ee56586de2b735ccf97b7f259ce47dab6e).
In 1.7.x, this issue doesn't occur.
